### PR TITLE
Custom errors, extending Error

### PR DESCRIPTION
**Pull request by mahdiHash**
Please review and comment/commit if anything is wrong.

### CODE REVIEW

File: 1-js/10-error-handling/2-custom-errors/article.md [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
+از درون، ما از `JSON.parse` استفاده خواهیم کرد. اگر این متد یک `json` ناقص را دریافت کند، سپس `SyntaxError` پرتاب می‌کند. اما اگر `json` از لحاظ سینتکس درست باشد به معنی یک کاربر معتبر نیست نه؟ ممکن است که داده مهم را نداشته باشد. برای مثال، ممکن است ویژگی‌های `name` و `age` که برای کاربران ما ضروری است را نداشته باشد.
 
-Our function `readUser(json)` will not only read JSON, but check ("validate") the data. If there are no required fields, or the format is wrong, then that's an error. And that's not a `SyntaxError`, because the data is syntactically correct, but another kind of error. We'll call it `ValidationError` and create a class for it. An error of that kind should also carry the information about the offending field.
+تابع `readUser(json)` نه تنها جی‌سان را می‌خواند بلکه داده را بررسی («اعتبارسنجی») می‌کند. اگر فیلدهای مورد نیاز وجود نداشته باشند یا شکل اشتباه باشد، پس یک ارور داریم. و این یک `SyntaxError` نیست چون داده از لحاظ سینتکس درست است بلکه نوع دیگری از ارور است. ما به آن `ValidationError` (ارور اعتبارسنجی) می‌گوییم و برای آن یک کلاس می‌سازیم. یک ارور از این نوع باید اطلاعاتی درباره فیلد متخلف را داشته باشد.
```
Comment (mahdiHash) [inline 1-js/10-error-handling/2-custom-errors/article.md:22]: More attention here, please.

File: 1-js/10-error-handling/2-custom-errors/article.md [hunk 2]
```diff
@@ -173,18 +173,18 @@ try {
...
+استفاده از کلاس جدید `PropertyRequiredError` آسان است: ما فقط باید اسم ویژگی را پاس دهیم: `new PropertyRequiredError(property)`. پیام `message` که برای انسان خوانا است توسط تابع سازنده تولید می‌شود.
 
-Please note that `this.name` in `PropertyRequiredError` constructor is again assigned manually. That may become a bit tedious -- to assign `this.name = <class name>` in every custom error class. We can avoid it by making our own "basic error" class that assigns `this.name = this.constructor.name`. And then inherit all our custom errors from it.
+لطفا توجه کنید که در تابع سازنده `PropertyRequiredError` مقدار `this.name` دوباره به صورت دستی مشخص می‌شود. این موضوع ممکن است کمی خسته‌کننده باشد -- مشخص کردن `this.name = <class name>` در هر کلاس شخصی‌سازی شده ارور. ما می‌توانیم با ایجاد کلاس «ارور پایه» خودمان که `this.name = this.constructor.name` را مشخص می‌کند از آن دوری کنیم. و سپس تمام ارور‌های شخصی‌سازی شده خودمان را از آن ارث‌بری کنیم.
 
-Let's call it `MyError`.
+بیایید به آن `MyError` بگوییم.
 
-Here's the code with `MyError` and other custom error classes, simplified:
+اینجا کد `MyError` و دیگر کلاس‌های ارور شخصی‌سازی شده را داریم، ساده شده:
```
Comment (mahdiHash) [inline 1-js/10-error-handling/2-custom-errors/article.md:187]: I'm not sure about translation of "simplified" here.

File: 1-js/10-error-handling/2-custom-errors/article.md [hunk 3]
```diff
@@ -205,51 +205,51 @@ class PropertyRequiredError extends ValidationError {
...
 ```
 
-In the code above we can see two types of errors, but there can be more.
+در کد بالا می‌توانیم دو نوع از ارور را ببینیم اما بیشتر از آن هم می‌تواند وجود داشته باشد.
 
-If the `readUser` function generates several kinds of errors, then we should ask ourselves: do we really want to check for all error types one-by-one every time?
+اگر تابع `readUser` چند نوع ارور تولید کند، سپس ما باید از خودمان بپرسیم: آیا واقعا می‌خواهیم هر بار برای تک تک ارورها بررسی انجام دهیم؟
 
-Often the answer is "No": we'd like to be "one level above all that". We just want to know if there was a "data reading error" -- why exactly it happened is often irrelevant (the error message describes it). Or, even better, we'd like to have a way to get the error details, but only if we need to.
+اغلب اوقات جواب «خیر» است: ما می‌خواهیم «یک پله بالاتر از تمام آن‌ها» باشیم. ما فقط می‌خواهیم بدانیم آیا یک «ارور خواندن داده» وجود داشت یا خیر -- اینکه دقیقا چرا اتفاق افتاد اغلب اوقات نامربوط است (پیام ارور این موضوع را توضیح می‌دهد). یا، حتی بهتر، می‌خواهیم راهی برای دریافت جزئیات ارور داشته باشیم اما فقط در صورتی که نیاز ما باشد.
```
Comment (mahdiHash) [inline 1-js/10-error-handling/2-custom-errors/article.md:242]: More attention here, please.

File: 1-js/10-error-handling/2-custom-errors/article.md (hunk 2 above)
Comment (mahdyar) [reply 1-js/10-error-handling/2-custom-errors/article.md:187]:
```suggestion
اینجا کد `MyError` و دیگر کلاس‌های ارور شخصی‌سازی شده را داریم، به صورت ساده‌شده:
```

**Review by mahdyar — CHANGES_REQUESTED** [review]
Hello and sorry for the delay;
Please review the suggested change(s) and request another review. :)

**Review by mahdyar — APPROVED** [review]

**Pull request merged**